spec/controllers/profile_controller_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe ProfileController, type: :controller do
+  before(:each) do
+    @request.host = '127.0.0.1:3000'
+    @user = User.create!(name: 'midd', email: 'user@example.org', password: 'very-secret')
+  end
+
+  context 'not get view' do
+    it 'edit role' do
+      sign_in @user
+      get :edit_role
+      expect(response).to render_template('edit_role')
+    end
+
+    it 'customer setup info' do
+      @user.role = :developer
+      @user.save!
+      sign_in @user
+      get :customer_setup_info
+      expect(response).to render_template('customer_setup_info')
+    end
+    it 'edit' do
+      sign_in @user
+      @user.role = :developer
+      @user.save!
+      get :edit
+      expect(response).to render_template('edit')
+    end
+  end
+
+  context 'not post to' do
+    it 'update' do
+      sign_in @user
+      @user.role = :developer
+      @user.save!
+      post 'update', params: { phone_number: '3423432', company_name: 'company_name user' }
+      expect(response).to redirect_to(root_url)
+    end
+
+    it 'update role' do
+      sign_in @user
+      post 'update_role', params: { role: :customer }
+      expect(response).to redirect_to(customer_setup_info_profile_path.to_s.split('?').first)
```

Comment:
Also check that params were actually saved

Reply:
done